R/liger.R

```diff
@@ -4038,25 +4064,91 @@ calcWilcoxon <- function(object, clusters = NULL, top.n = 10){
+#' Subset a dataframe produced by calculateWilcox by feature, dataset, or column
+#' 
+#' @param object \code{liger} object. Should run calculateWilcox first 
+#' @param features A list of features to include in the subset (default NULL)
+#' @param datasets A list of datasets to include in the subset  (defualt NULL).
+#' @param columns A list of columns to include in the subset  (default NULL).
+#' @param return.subset Return the subsetted dataframe instead of a liger object (default FALSE)
+#' @param save.subset Replace the unsubset dataframe with the subset (default TRUE)
+#'
+#' @return Updated \code{liger} object.
+#' @export
+#' @importFrom presto wilcoxauc
+#' @examples
+#' \dontrun{
+#' # liger object, after running CalculateWilcoxon 
+#' ligerex
+#' # create lists of features, datasets, and columns to subset by
+#' genes = c("UBBP4","KIAA1715","TMEM50B")
+#' datasets = c("tenx")
+#' #
+#' ligerex <- subsetWilcoxon(ligerex, features = genes, datasets = datasets)
+#' }
+subsetWilcoxon <- function(object, features = NULL, datasets = NULL, columns = NULL,
+                           return.subset = FALSE, save.subset = TRUE){
+  if(!return.subset && !save.subset){
+    return(object)
+  }
+  if(!is.null(columns)){
+    columns = c("feature", "group", columns)
+    wilcoxon = object@wilcoxon[columns]
+  }
+  if(!is.null(features)){
+    wilcoxon = object@wilcoxon[object@wilcoxon$feature == toupper(features)]
+  }
+  if(!is.null(datasets)){
+    for(name in datasets){
+      wilcoxon = object@wilcoxon[grepl(toupper(name), object@wilcoxon$group, fixed = TRUE)]
+    }
+  }
+  if(save.subset){
+    object@wilcoxon = wilcoxon
+  }
+  if(return.subset){
+    return(wilcoxon)
+  }
+  return(object)
+}
+
+#wrapper for top marker
+listMarkersWilcoxon <- function(object, ...){
+  return(top_markers(object@wilcoxon, ...))
+}
+
+
+#' Make Jaccard matrix (will have use when imputation is implemented)
+#' 
+#' @param object \code{liger} object. Should run scaleNotCenter first 
+#'
+#' @return Updated \code{liger} object.
+#' @export
+#' @examples
+#' \dontrun{
+#' # liger object, after running scaleNotCenter 
+#' ligerex
+#' # add Jaccard matrix to a list for each dataset
+#' ligerex <- runJaccard(ligerex)
+#' }
 runJaccard <- function(object){
-    for(i in 1:length(object@scale.data)){
-      common_values <- object@scale.data[[i]] %*% t(object@scale.data[[i]])
+    for(i in object@scale.data){
+      common_values <- i %*% t(i)
       nonzero_value_index <- which(common_values > 0,  arr.ind=TRUE)
       nonzero_value <- common_values[nonzero_value_index]
-      row_sums <- rowSums(t(object@scale.data[[i]]))
+      row_sums <- rowSums(t(i))
       jaccard <- nonzero_value/(row_sums[nonzero_value_index[,1]]
                                 +row_sums[nonzero_value_index[,1]] - nonzero_value)
-      object@jaccard[[i]] <- sparseMatrix(i = nonzero_value_index[,1],
+      object@jaccard.append(sparseMatrix(i = nonzero_value_index[,1],
```

Comment:
`.append` is not a native function of lists in R (as much as I wish R were like Python...). Do you have some dependency you're planning to use here? Also, try not to include incomplete functions that will break the unit tests in a commit along with other functions. It makes it difficult to test the complete functions as it prevents the package from installing. Try separating them into different commits.